homeassistant/components/nanoleaf/config_flow.py

```diff
@@ -0,0 +1,272 @@
+        return await self._async_discovery_handler(discovery_info)
+
+    async def _async_discovery_handler(
+        self, discovery_info: DiscoveryInfoType
+    ) -> FlowResult:
+        """Handle Nanoleaf discovery."""
+        host = discovery_info["host"]
+        name = discovery_info["name"].replace(f".{discovery_info['type']}", "")
+        await self.async_set_unique_id(name)
+        self._abort_if_unique_id_configured({CONF_HOST: host})
+        self.nanoleaf = Nanoleaf(host)
+
+        # Import from discovery integration
+        self.device_id = discovery_info["properties"]["id"]
+        self.discovery_conf = dict(load_json(self.hass.config.path(CONFIG_FILE)))
+        self.nanoleaf.token = self.discovery_conf.get(host, {}).get("token")  # < 2021.4
+        self.nanoleaf.token = self.discovery_conf.get(self.device_id, {}).get(
+            "token", self.nanoleaf.token
+        )  # >= 2021.4
+        if self.nanoleaf.token is not None:
+            self.context["source"] = config_entries.SOURCE_INTEGRATION_DISCOVERY
+            _LOGGER.warning(
+                "Importing Nanoleaf %s from the discovery integration", name
+            )
+            return await self.async_setup_finish()
+
+        self.context["title_placeholders"] = {"name": name}
+        return self.async_show_form(step_id="link")
+
+    async def async_step_link(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle Nanoleaf link step."""
+        if user_input is None:
+            return self.async_show_form(step_id="link")
+
+        try:
+            await self.hass.async_add_executor_job(self.nanoleaf.authorize)
+        except NotAuthorizingNewTokens:
+            _LOGGER.warning("Nanoleaf is not allowing new tokens")
+            return self.async_show_form(
+                step_id="link", errors={"base": "not_allowing_new_tokens"}
+            )
+        except Unavailable:
+            _LOGGER.error("Nanoleaf is not available at %s", self.nanoleaf.host)
+            return self.async_show_form(
+                step_id="user",
+                data_schema=self.user_data_schema,
+                errors={"base": "cannot_connect"},
+                last_step=False,
+            )
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unknown error authorizing Nanoleaf")
+            return self.async_show_form(step_id="link", errors={"base": "unknown"})
+        return await self.async_setup_finish()
+
+    async def async_step_manual(self, user_input: dict[str, Any]) -> FlowResult:
+        """Handle Nanoleaf manual step."""
+        self.nanoleaf.token = user_input[CONF_TOKEN]
+        return await self.async_setup_finish()
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Handle Nanoleaf configuration import."""
+        self._async_abort_entries_match({CONF_HOST: config[CONF_HOST]})
+        _LOGGER.warning(
```

Comment:
Remove this warning. 
```suggestion
        _LOGGER.debug(
```